Remove mention of "left-hand side expression" from operators guide (#18842)
mdn/content: files/en-us/web/javascript/guide/expressions_and_operators/index.md

This chapter describes JavaScript's expressions and operators, including assignment, comparison, arithmetic, bitwise, logical, string, ternary and more.

> An _expression_, at a high level, is any valid unit of code that resolves to a value. Every syntactically valid expression resolves to some value — but conceptually, there are two types of expressions: those that have side effects (such as assigning values), and those that purely _evaluate_.

Reviewer: ```suggestion
At a high level, an _expression_ is a valid unit of code that resolves to a value. There are two types of expressions: those that have side effects (such as assigning values) and those that purely _evaluate_.
```

The expression `x = 7` is an example of the first type. This expression uses the `=` _operator_ to assign the value seven to the variable `x`. The expression itself evaluates to `7`.

> The expression `3 + 4` is an example of the second type. This expression uses the `+` operator to add `3` and `4` together, and produces a value `7`. However, if it's not eventually part of a bigger construct (for example, a [variable declaration](/en-US/docs/Web/JavaScript/Guide/Grammar_and_Types#declarations)), its result will be immediately discarded — this is usually a programmer mistake, because the evaluation doesn't produce any effects.

Reviewer: ```suggestion
The expression `3 + 4` is an example of the second type. This expression uses the `+` operator to add `3` and `4` together and produces a value, `7`. However, if it's not eventually part of a bigger construct (for example, a [variable declaration](/en-US/docs/Web/JavaScript/Guide/Grammar_and_Types#declarations) like `let z = 3 + 4`), its result will be immediately discarded — this is usually a programmer mistake because the evaluation doesn't produce any effects.
```

As the examples above also illustrate, all complex expressions are joined by _operators_, such as `=` and `+`. In this section, we will introduce the following operators:

- [Assignment operators](#assignment_operators)
- [Comparison operators](#comparison_operators)
- [Arithmetic operators](#arithmetic_operators)
- [Bitwise operators](#bitwise_operators)
- [Logical operators](#logical_operators)
- [BigInt operators](#bigint_operators)
- [String operators](#string_operators)
- [Conditional (ternary) operator](#conditional_ternary_operator)
- [Comma operator](#comma_operator)
- [Unary operators](#unary_operators)
- [Relational operators](#relational_operators)

These operators join operands either formed by higher-precedence operators or one of the [basic expressions](#basic_expressions). A complete and detailed list of operators and expressions is also available in the [reference](/en-US/docs/Web/JavaScript/Reference/Operators).

The _precedence_ of operators determines the order they are applied when evaluating an expression. Fo example:

```js
const x = 1 + 2 * 3;
const y = 2 * 3 + 1;
```

> Despite `*` and `+` coming in different orders, both expressions would result in `7`, because `*` has higher precedence than `+`, so the `*`-joined expression will always be evaluated first. You can override operator precedence by using parentheses (which create a [grouped expression](#grouping_operator) — the basic expression). To see a complete table of operator precedence as well as various caveats, see the [Operator Precedence Reference](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table) page. 

Reviewer: ```suggestion
Despite `*` and `+` coming in different orders, both expressions would result in `7` because `*` has precedence over `+`, so the `*`-joined expression will always be evaluated first. You can override operator precedence by using parentheses (which creates a [grouped expression](#grouping_operator) — the basic expression). To see a complete table of operator precedence as well as various caveats, see the [Operator Precedence Reference](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table) page. 
```

Author: "Parentheses" is plural and "using parentheses" is singular... I guess I'm fine with both

JavaScript has both _binary_ and _unary_ operators, and one special ternary operator, the conditional operator.
A binary operator requires two operands, one before the operator and one after the operator:

```
operand1 operator operand2
```

> For example, `3+4` or `x*y`. This form is called an _infix_ binary operator, because the operator is placed between two operands. All binary operators in JavaScript are infix.

Reviewer: ```suggestion
For example, `3 + 4` or `x * y`. This form is called an _infix_ binary operator, because the operator is placed between two operands. All binary operators in JavaScript are infix.
```

A unary operator requires a single operand, either before or after the operator:

```
operator operand
operand operator
```

> For example, `x++` or `++x`. The `operator operand` form is called a _postfix_ unary operator, and the `operand operator` form is called a _prefix_ unary operator. `++` and `--` are the only postfix operators in JavaScript — all other operators, like `!`, `typeof`, etc. are prefix.

Reviewer: ```suggestion
For example, `x++` or `++x`. The _operator operand_ form is called a _postfix_ unary operator, and the _operand operator_ form is called a _prefix_ unary operator. `++` and `--` are the only postfix operators in JavaScript — all other operators, like `!`, `typeof`, etc. are prefix.
```

Author: It's actually meant to be code, though. (We are substituting `operand` and `operator` with other tokens.) Also since it's quoting the code block above.

Reviewer: Let's keep then.
